appcues/src/main/java/com/appcues/AnalyticsPublisher.kt

```diff
@@ -0,0 +1,69 @@
+package com.appcues
+
+import com.appcues.AnalyticType.EVENT
+import com.appcues.AnalyticType.GROUP
+import com.appcues.AnalyticType.IDENTIFY
+import com.appcues.AnalyticType.SCREEN
+import com.appcues.analytics.ActivityRequestBuilder
+import com.appcues.analytics.TrackingData
+import com.appcues.data.remote.request.EventRequest
+import java.util.Date
+
+internal class AnalyticsPublisher(
+    private val storage: Storage
+) {
+
+    fun publish(listener: AnalyticsListener?, data: TrackingData) {
+        if (listener == null) return
+
+        when (data.type) {
+            EVENT -> data.request.events?.forEach {
+                listener.trackedAnalytic(EVENT, it.name, it.attributes.sanitize(), data.isInternal)
+            }
+            IDENTIFY -> listener.trackedAnalytic(IDENTIFY, storage.userId, data.request.profileUpdate, data.isInternal)
+            GROUP -> listener.trackedAnalytic(GROUP, storage.groupId, data.request.groupUpdate, data.isInternal)
+            SCREEN -> data.request.events?.forEach {
+                listener.trackedAnalytic(SCREEN, it.screenTitle(), it.attributes.sanitize(), data.isInternal)
+            }
+        }
+    }
+
+    private fun EventRequest.screenTitle(): String? =
+        attributes[ActivityRequestBuilder.SCREEN_TITLE_ATTRIBUTE] as? String
+
+    private fun Map<*, *>.sanitize(): MutableMap<String, Any> {
+        val sanitizedMap = mutableMapOf<String, Any>()
+
+        forEach {
+            val key = it.key
+            val value = it.value
+            if (key is String && value != null) {
```

Comment:
This is where we'll still need something to format the formState as a map first, like https://github.com/appcues/appcues-ios-sdk/blob/main/Sources/AppcuesKit/Data/Analytics/AnalyticsBroadcaster.swift#L50-L54